rpd_generator/bdl_structure/bdl_commands/system.py

```diff
@@ -1330,6 +1394,73 @@ def populate_air_energy_recovery(self):
                 self.air_energy_recovery_outdoor_airflow
             )
 
+    def populate_fan_operation_during_occupied(self):
+        fan_sch_name = self.keyword_value_pairs.get(BDL_SystemKeywords.FAN_SCHEDULE)
+
+        if (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.RESVVT
+        ):
+            return FanSystemOperationOptions.CYCLING
+
+        elif (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.DOAS
+        ):
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            if not fan_sch:
+                return FanSystemOperationOptions.INTERMITTENT
+            else:
+                return FanSystemOperationOptions.CONTINUOUS
+
+        elif self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE) in [
+            BDL_SystemTypes.PTAC,
+            BDL_SystemTypes.HP,
+            BDL_SystemTypes.UVT,
+        ]:
+            if not self.fan_sys_minimum_outdoor_airflow:
+                return FanSystemOperationOptions.CYCLING
+
+            min_oa_sch_name = self.keyword_value_pairs.get(
+                BDL_SystemKeywords.MIN_AIR_SCH
+            )
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            min_oa_sch = self.rmd.bdl_obj_instances.get(min_oa_sch_name)
+            if not min_oa_sch:
+                return FanSystemOperationOptions.CONTINUOUS
+            if not fan_sch:  # (and min_oa_sch)
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if min_oa_sch_values:
+                    for i in range(len(min_oa_sch_values)):
+                        if min_oa_sch_values[i] == 0:
+                            return FanSystemOperationOptions.CYCLING
+                    return FanSystemOperationOptions.CONTINUOUS
+                return FanSystemOperationOptions.CYCLING
```

Comment:
I am not following the reason for returning cycling here

Reply:
Really this is an edge case in the logic that I do not expect will ever occur. It should return an error message that the min oa hourly values could not be interpreted instead of returning cycling